### DISCUSSION TITLE
Setup by old release [help]

### DISCUSSION THREAD

**Discussion by davirs** [markshust/docker-magento#735] [Q&A]
Hi! I can read the tutorial using the curl instalation and all things worked fine.

How can I install release 39.0.2 , for example? I never did it before.

Thank you

**Comment by mauriciomueller** [comment]
The curl installation just runs some commands that are listed in https://raw.githubusercontent.com/markshust/docker-magento/master/lib/template link. If you access that link in your browser you will see these commands.

I think you can download the zip file from releases tag [release that you want] and do all the commands manually.

I didn't test, but try something like this:

    First download the zip file: https://github.com/markshust/docker-magento/releases/tag/39.0.2
    Unzip it to your project folder
    Run the command: mv compose/* ./
    Run the command: mv compose/.gitignore ./
    Run the command: mv compose/.vscode ./
    Run the command: rm -rf compose .git

Note:
Ensure these are created so Docker doesn't create them as root

mkdir -p ~/.composer ~/.ssh

**Reply by davirs** [reply]
@mauriciomueller thank you!

I runned the command bin/download magento 2.4.0 (this is the project majento version) after prepare de folder with your help.

so I got this error:

`Problem 1
    - Root composer.json requires dealerdirect/phpcodesniffer-composer-installer ^0.5.0 -> satisfiable by dealerdirect/phpcodesniffer-composer-installer[v0.5.0].
    - dealerdirect/phpcodesniffer-composer-installer v0.5.0 requires composer-plugin-api ^1.0 -> found composer-plugin-api[2.1.0] but it does not match the constraint.
  Problem 2
    - laminas/laminas-dependency-plugin[1.0.0, ..., 1.0.4] require composer-plugin-api ^1.1 -> found composer-plugin-api[2.1.0] but it does not match the constraint.
    - magento/product-community-edition 2.4.0-p1 requires laminas/laminas-dependency-plugin ^1.0 -> satisfiable by laminas/laminas-dependency-plugin[1.0.0, ..., 1.0.4].
    - Root composer.json requires magento/product-community-edition 2.4.0-p1 -> satisfiable by magento/product-community-edition[2.4.0-p1].`
    
   Can you tell me I will need solve this items one by one or I must try another release?
   
   thank you

**Reply by mauriciomueller** [reply]
I think you are having a problem because of the PHP version. It may be set at 8.0.

Try to change your php-fpm version at docker-composer.yml file.

If your containers are running, run bin/stop in order to stop all containers, because you won't be able to stop them after changing the docker-composer.yml file.

Then edit the docker-composer.yml file changing the image that is set at services phpfpm block.

    phpfpm:
    image: markoshust/magento-php:7.4-fpm-15
    volumes: *appvolumes
    env_file: env/phpfpm.env
    #extra_hosts: *appextrahosts

---- Remember to indent it -----

Now run bin/start.

Then you can run the composer command.

There are 2 ways to do it.

bin/composer install

or

bin/bash to access the bash inside the container and run composer install.

There is also some discussion about that fallback here: https://github.com/markshust/docker-magento/issues/685, but I think my approach will be quicker for your case/stage.

Let me know if it worked.
